packages/panzer/disc-fe/src/Panzer_IntegrationValues2.cpp

```diff
@@ -59,47 +59,629 @@
+          // We can do this by setting t = n_0 \times n_1
+          PANZER_CROSS(n0,n1,t);
+
+          // Normalize the transverse vector
+          const Scalar mag_t = Kokkos::Experimental::sqrt(PANZER_DOT(t,t));
+          t[0] /= mag_t;
+          t[1] /= mag_t;
+          t[2] /= mag_t;
+
+          // The binormal will be the sum of the normals (does not need to be a right handed system)
+          b[0] = n0[0] + n1[0];
+          b[1] = n0[1] + n1[1];
+          b[2] = n0[2] + n1[2];
+
+          // Normalize the binormal vector
+          const Scalar mag_b = Kokkos::Experimental::sqrt(PANZER_DOT(b,b));
+          b[0] /= mag_b;
+          b[1] /= mag_b;
+          b[2] /= mag_b;
+
+        }
+      }
+
+      // Now that we have a reference coordinate plane in which to align our points
+      // Point on the transverse/binormal plane
+      Scalar p0[2] = {0};
+      Scalar p1[2] = {0};
+
+      // Differential position in mesh
+      Scalar x0[3] = {0};
+      Scalar x1[3] = {0};
+
+      // Iterate through points in cell 1 and find which point they align with in cell 0
+      for(int face_point_1=0; face_point_1<num_points_per_face; ++face_point_1){
+
+        // Get the point index in the 1 cell
+        const int point_1 = lidx_1*num_points_per_face + face_point_1;
+
+        // Load position shifted by face center
+        for(int dim=0; dim<cell_dim; ++dim)
+          x1[dim] = surface_points(cell_1,point_1,dim) - xc1[dim];
+
+        // Convert position to transverse/binormal plane
+        p1[0] = PANZER_DOT(x1,t);
+        p1[1] = PANZER_DOT(x1,b);
+
+        // Compare to points on the other surface
+        for(int face_point_0=0; face_point_0<num_points_per_face; ++face_point_0){
+
+          // Get the point index in the 0 cell
+          const int point_0 = lidx_0*num_points_per_face + face_point_0;
+
+          // Load position shifted by face center
+          for(int dim=0; dim<cell_dim; ++dim)
+            x0[dim] = surface_points(cell_0,point_0,dim) - xc0[dim];
+
+          // Convert position to transverse/binormal plane
+          p0[0] = PANZER_DOT(x0,t);
+          p0[1] = PANZER_DOT(x0,b);
+
+          // Find the distance squared between p0 and p1
+          const Scalar p012 = (p0[0]-p1[0])*(p0[0]-p1[0]) + (p0[1]-p1[1])*(p0[1]-p1[1]);
+
+          // TODO: Should this be a constant value, or should we just find the minimum point?
+          // If the distance, compared to the size of the cell, is small, we assume these are the same points
+          if(p012 / r2 < 1.e-12){
+//            permutation(cell_1,point_1) = lidx_1*num_points_per_face+face_point_0;
+            permutation(cell_1,lidx_1*num_points_per_face+face_point_0) = point_1;
+            break;
+          }
+
+          // Big problem - there wan't a point that aligned properly
+//          KOKKOS_ASSERT(face_point_0 != num_points_per_face-1);
```

Comment:
Does this need to be uncommented?

Reply:
Removed. It was left over from when I copy/pasted it from elsewhere. It doesn't really apply any more.